src/main/java/com/khutircraftubackend/auth/AuthenticationService.java

```diff
@@ -51,70 +52,129 @@ public class AuthenticationService {
      * @throws BadCredentialsException Якщо вказані невірні дані для входу.
      * @throws AuthenticationException Якщо виникла непередбачена помилка під час аутентифікації.
      */
+
     @Transactional
     public AuthResponse authenticate(LoginRequest request) {
-        authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(
-                        request.email(),
-                        request.password())
-        );
-
-        // Fetch user by email
-        UserEntity user = userRepository.findByEmail(request.email())
-                .orElseThrow(() -> new UserNotFoundException("Користувач з таким email не знайдений"));
-
-        if (Boolean.FALSE.equals(user.isEnabled())) {
-            return AuthResponse
-                    .builder()
-                    .message("Підтвердження за поштою не пройдено")
-                    .build();
+
+        UserEntity user = getUserForEmail(request.email());
+        if (!user.isEnabled()) {
+            return buildDisabledUser(user);
         }
-        String jwt = jwtUtils.generateJwtToken(user.getEmail());
+
+        authenticateUser(request.email(), request.password());
+        String token = jwtUtils.generateJwtToken(user.getEmail());
+        return buildEnabledUser(user, token);
+    }
+
+    private UserEntity getUserForEmail(String email) {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UserNotFoundException(String.format(AuthResponseMessages.EMAIL_NOT_FOUND, email)));
+    }
+
+    private void authenticateUser(String email, String password) {
+        UsernamePasswordAuthenticationToken authenticationToken = new UsernamePasswordAuthenticationToken(email, password);
+        authenticationManager.authenticate(authenticationToken);
+    }
+
+    private AuthResponse buildDisabledUser(UserEntity user) {
+        return AuthResponse
+                .builder()
+                .message(String.format(AuthResponseMessages.EMAIL_DISABLED, user.getEmail()))
+                .build();
+    }
+
+    private AuthResponse buildEnabledUser(UserEntity user, String token) {
         return AuthResponse.builder()
-                .jwt(jwt)
+                .jwt(token)
                 .email(user.getEmail())
-                .message("Вітаємо з успішним входом")
+                .message(String.format(AuthResponseMessages.EMAIL_ENABLED))
                 .build();
     }
 
     @Transactional
     @PostUpdate
     public AuthResponse registerNewUser(RegisterRequest request) {
-        if(userRepository.existsByEmail(request.email())) {
-            throw new UserExistsException("Цей email зайнятий, використайте інший або ввійдіть під цим");
+
+        validateEmailNotInUse(request.email());
```

Comment:
validateEmailNotInUse has only one usage,
no need in taking it out into an external method, unless it needs to be reused.

Reply:
we can just apply `fail-fast` approach here and check at the very beginning user exists, and throw exception if so.